Sources/StreamChatUI/Utils/VideoPreviewLoader.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import AVKit
+import StreamChat
+import UIKit
+
+public protocol VideoPreviewLoader: AnyObject {
+    func loadPreviewForVideo(at url: URL, completion: @escaping (Result<UIImage, Error>) -> Void)
+}
+
+final class DefaultVideoPreviewLoader: VideoPreviewLoader {
+    @Atomic private var cache = Cache<URL, UIImage>()
+    
+    init() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleMemoryWarning(_:)),
+            name: UIApplication.didReceiveMemoryWarningNotification,
+            object: nil
+        )
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    func loadPreviewForVideo(at url: URL, completion: @escaping (Result<UIImage, Error>) -> Void) {
+        if let cached = cache[url] {
+            return completion(.success(cached))
+        }
+        
+        let asset = AVURLAsset(url: url)
+        let imageGenerator = AVAssetImageGenerator(asset: asset)
+        let frameTime = CMTime(seconds: 1, preferredTimescale: 600)
+        
+        imageGenerator.generateCGImagesAsynchronously(forTimes: [.init(time: frameTime)]) { [weak self] _, image, _, _, error in
+            let result: Result<UIImage, Error>
+            if let thumbnail = image {
+                result = .success(.init(cgImage: thumbnail))
+            } else if let error = error {
+                result = .failure(error)
+            } else {
+                log.error("Both error and image are `nil`.")
+                return
+            }
+            
+            self?._cache.mutate {
+                $0[url] = try? result.get()
+            }
```

Comment:
This is not need since NSCache is thread-safe like I said above 👍

Reply:
✅ Fixed